ouroboros-consensus/src/Ouroboros/Consensus/MiniProtocol/ChainSync/Server.hs

```diff
@@ -43,7 +43,7 @@ chainSyncHeadersServer
     -> ChainDB m blk
     -> NodeToNodeVersion
     -> ResourceRegistry m
-    -> ChainSyncServer (SerialisedHeader blk) (Tip blk) m ()
+    -> ChainSyncServer (SerialisedHeader blk) (Point (SerialisedHeader blk)) (Tip blk) m ()
```

Comment:
If you prefer, we can turn this into `Point blk` with some calls to `castPoint` in the right places.